### PR TITLE
add helper fun for Padding

### PR DESCRIPTION
**Pull request by wiiznokes**
This add some new method to the Padding struct. 

I have some other ideas, like also provide some more method like

```rust 
/// Create a [`Padding`] with equally spaces of the right and left side
    pub fn vertical(padding: f32) -> Self {
        Self {
            left: padding,
            right: padding,
            ..Padding::ZERO
        }
    }
```
and the same for vertical, but that will involve breaking change.

And also provide a constructor API alternative:

```rust 
pub fn vertical(self, padding: f32) -> Self {
        self.left = padding;
        self.right = padding;
        self
    }
```

to allow writting this:

```rust 
let padding = Padding::horizontal(10f32)
            .bottom(5f32);
```

But this will involve, reanaming the method i written like such

```rust 
fn new_top()
fn new_bottom()
fn new_vertical()
```

which is fine imo.

I think it make more sense to provide this funcions rather than

```rust
pub fn vertical(self) -> f32 {
        self.top + self.bottom
    }
```

because i don't really see a lot of use case for this (user perspective).

### CODE REVIEW

File: core/src/padding.rs [hunk 1]
```diff
@@ -64,6 +64,38 @@ impl Padding {
         }
     }
 
+    /// Create a top Padding
```
Comment (hecrj) [inline core/src/padding.rs:67]: Let's use the link notation and end with a period:

```rust
/// Create a top [`Padding`].
```

**Review by hecrj — CHANGES_REQUESTED** [review]

**Comment by Koranir** [comment]
I think that instead of these new function being constructors, they should be builder-like functions. It makes the api a lot more flexible, at the small cost of turning `Padding::bottom(5.0)` into `Padding::ZERO.bottom(5.0)`.

**Comment by Koranir** [comment]
E.g.
```rust
    /// Sets the top padding to `padding`.
    pub const fn top(self, padding: f32) -> Self {
        Self {
            top: padding,
            ..self
        }
    }
```

**Comment by wiiznokes** [comment]
> I think that instead of these new function being constructors, they should be builder-like functions. It makes the api a lot more flexible, at the small cost of turning Padding::bottom(5.0) into Padding::ZERO.bottom(5.0).

Yes that what i was saying. But we can still use `Padding::new_bottom(5.0)` for constructor and `padding.bottom(5.0)` for builder.

What do you think @hecrj ?

**Comment by Koranir** [comment]
> > I think that instead of these new function being constructors, they should be builder-like functions. It makes the api a lot more flexible, at the small cost of turning Padding::bottom(5.0) into Padding::ZERO.bottom(5.0).
> 
> Yes that what i was saying. But we can still use `Padding::new_bottom(5.0)` for constructor and `padding.bottom(5.0)` for builder.
> 
> What do you think @hecrj ?

The naming convention for constructors of this type is `with_<thing>`, so `Padding::with_bottom(5.0)` in this example. However, I don't think having separate constructors for each is required here and just adds unnecessary complexity.

**Comment by wiiznokes** [comment]
> However, I don't think having separate constructors for each is required here and just adds unnecessary complexity.

I better than

```rust
Padding {
   bottom: 5.0,
   ..Padding::ZERO
}
```
(one line instead of four)

**Comment by Koranir** [comment]
> > However, I don't think having separate constructors for each is required here and just adds unnecessary complexity.
> 
> I better than
> 
> ```rust
> Padding {
>    bottom: 5.0,
>    ..Padding::ZERO
> }
> ```
> 
> (one line instead of four)

With builder pattern it would be
```rust
Padding::ZERO.bottom(5.0)
```

**Comment by wiiznokes** [comment]
> With builder pattern it would be
> 
> ```rust
> Padding::ZERO.bottom(5.0)
> ```

True, still prefer the `with_bottom` though.

**Comment by wiiznokes** [comment]
we could also add `with_vertical`, `with_horizontal` and change the behavior of `vertical` and `horizontal`.

**Comment by wiiznokes** [comment]
I think we could also remove the From impl

**Pull request closed without merging**